PutAndVerify/WriteAndVerify.tla

```diff
@@ -0,0 +1,93 @@
+--------------------------- MODULE WriteAndVerify ---------------------------
+
+CONSTANT CLIENTS, WITH_RO, WITH_CRASHING
+
+VARIABLE storage, clState, clFavor
+
+Vars == << storage, clState, clFavor >>
+
+WVTypeOk == /\ storage \in [ CLIENTS -> { "none", "empty", "written", "written-ro", "empty-ro" } ]
+            /\ clState \in [ CLIENTS -> { "begin", "precheck-done", "blank-written", 
+                                          "write-check-done", "committed", "aborted", "crashed" } ]
+            /\ clFavor \in [ CLIENTS -> { "read-write", "read-only" } ]
+
+WVInit == /\ storage = [ cli \in CLIENTS |-> "none" ]
+          /\ clState = [ cli \in CLIENTS |-> "begin" ]
+          /\ clFavor = [ cli \in CLIENTS |-> "read-write" ]
+
+(* Only one execution can be committed. *)
+(* Once committed, the file has been written. *)
+WVConsistency == /\ \A i, j \in CLIENTS : 
+                    i /= j /\ storage[i] = "written" => ~storage[j] \in { "written", "written-ro" }
+
+(* "Verify" condition: storage only contains our file or nothing. *)
+Check(c) == /\ clFavor[c] = "read-write" =>
+               \A f \in CLIENTS : f /= c => storage[f] = "none"
+            /\ clFavor[c] = "read-only" =>
+               (* No conflicting write. *)
+                \A f \in CLIENTS : f /= c => ~storage[f] \in { "empty", "written" }  

```

Comment:
can you explain in comment why other clients' storrage can't be "empty"? I didn't get  the meaning of this property